spec/app/domain/responses_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe SuccessResponse do
```

Comment:
Maybe the unit tests for `[Success|Failure]Response` could confirm the behavior of `.bind`. I spent the most time digesting `[Success|Failure]Response.bind`, any a unit test case or two might've made their behavior plain to see.

Reply:
Good call.  I'll add some tests.  `bind` isn't the most intuitive action.